pkg/provider/provider.go

```diff
@@ -107,12 +107,80 @@ func makeKubeProvider(
 
 // CheckConfig validates the configuration for this provider.
 func (k *kubeProvider) CheckConfig(ctx context.Context, req *pulumirpc.CheckRequest) (*pulumirpc.CheckResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "CheckConfig is not yet implemented")
+	// TODO: Check for failures if needed.
+	return &pulumirpc.CheckResponse{Inputs: req.GetNews()}, nil
```

Comment:
I wonder, do we need to start returning "defaults" here?  For example, if you are using the default provider and not explicitly setting a cluster or context or kubeconfig in your stack's configuration, I think this is going to return an empty set of inputs.  If later, you go and change the kubeconfig by hand and re-run up, we won't see any changes.  Is that actually the behavior we want here?

Reply:
If we do that, we should be careful about cases where in DiffConfig the "olds" are empty, since when upgrading to a newer CLI and a provider with this change, the initial call to DiffConfig will more or less always pass in an empty olds bag.

Reply:
>  (e.g. I think the AWS Go SDK will pick up certain things that are not reflected in the provider config; @jen20 may be able to confirm).

Right, I agree that what we have here is similar to the default behavior for TF based providers, my question was more about the UX we /want/ to have.

Reply:
Yes, the `config` is a combination of explicitly set values and what we pick up from `awsbase.Credentials` and our own provider preconfigure call back.